Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Lights/Ltc.azsli

```diff
@@ -80,14 +80,44 @@ float3x3 BuildViewAlignedOrthonormalBasis(in float3 normal, in float3 dirToView)
 // xy plane in positive z space.
 float IntegrateEdge(float3 v1, float3 v2)
 {
+    // This alternate version may work better for platforms where acos() precision is low.
+    /*
+    float x = dot(v1, v2);
+    float y = abs(x);
+
+    float a = 5.42031 + (3.12829 + 0.0902326 * y) * y;
+    float b = 3.45068 + (4.18814 + y) * y;
+    float theta_sinTheta = a / b;
+
+    if (x < 0.0)
+    {
+        theta_sinTheta = PI * rsqrt(saturate(1.0 - x * x)) - theta_sinTheta;
+    }
+
+    float3 u = cross(v1, v2);
+    return theta_sinTheta * u.z;
+    */
+
     float cosTheta = dot(v1, v2);
-    cosTheta = clamp(cosTheta, -0.9999, 0.9999);
+    float theta = acos(cosTheta);
 
     // calculate 1.0 / sin(theta)
-    float invSinTheta = rsqrt(1.0 - cosTheta * cosTheta);
+    float invSinTheta = rsqrt(saturate(1.0 - cosTheta * cosTheta));
 
-    float theta = acos(cosTheta);
-    return cross(v1, v2).z * theta * invSinTheta;
+    return cross(v1, v2).z * ((theta > 0.001) ? theta * invSinTheta : 1.0);
+}
+
+// Cheaper version of above which is good enough for diffuse
+float IntegrateEdgeDiffuse(float3 v1, float3 v2)
+{
+    float cosTheta = dot(v1, v2);
+    float absCosTheta = abs(cosTheta);
+    float theta_sinTheta = 1.5708 + (-0.879406 + 0.308609 * absCosTheta) * absCosTheta;
+    if (cosTheta < 0.0)
+    {
```

Comment:
Looking at the more advanced solution above, why not have theta_sinTheta be simply 1 below the horizon?

Reply:
It's not for below horizon. cosTheta will vary from -1 to 1, and both points can be above the hemisphere (imagine two points on the horizon opposite each other.

In the "advanced" version, the theta > 0.001 is meant as protection from crazy values as v1 and v2 get very close to each other. This leads to a cosTheta of 1, and a theta of 0, which would mean invSinTheta would be the inverse square root of 0... which is problematic. However the limit of the equation as cosTheta approches 1 is 1, so that's why we just say 1 when the value is close enough.